common/frame.go

```diff
@@ -315,27 +315,51 @@ func (f *Frame) defaultTimeout() time.Duration {
 func (f *Frame) document() (*ElementHandle, error) {
 	f.log.Debugf("Frame:document", "fid:%s furl:%q", f.ID(), f.URL())
 
-	rt := k6common.GetRuntime(f.ctx)
-	if f.documentHandle != nil {
-		return f.documentHandle, nil
+	if cdh, ok := f.cachedDocumentHandle(); ok {
+		return cdh, nil
 	}
 
 	f.waitForExecutionContext(mainWorld)
 
+	dh, err := f.newDocumentHandle()
+	if err != nil {
+		return nil, fmt.Errorf("newDocumentHandle: %w", err)
+	}
+	f.executionContextMu.Lock()
+	defer f.executionContextMu.Unlock()
+	f.documentHandle = dh
```

Comment:
Lock for minimizing the `documentHandle` update time. Like I said in the previous comment, it may offer a chance for getting fresh document handles.

Reply:
This is not intuitive, and I'm still not sure I get why locking `f.executionContextMu` would address it, but can you add a comment for it?

Reply:
> I wasn't able to reproduce the `TestStaleExecutionContext` failure.

I hit it after a couple of tries with this:

```shell
cd tests
stress.bash -test.run=TestStaleExecutionContext -test.v

==================
WARNING: DATA RACE
Write at 0x00c000001b08 by goroutine 65:
  github.com/grafana/xk6-browser/common.(*Frame).nullContext()
      /Users/inanc/grafana/k6b/common/frame.go:384 +0x3bc
  github.com/grafana/xk6-browser/common.(*FrameSession).onExecutionContextsCleared()
      /Users/inanc/grafana/k6b/common/frame_session.go:611 +0x284
  github.com/grafana/xk6-browser/common.(*FrameSession).initEvents.func1()
      /Users/inanc/grafana/k6b/common/frame_session.go:277 +0x6a4

Previous read at 0x00c000001b08 by goroutine 24:
  github.com/grafana/xk6-browser/common.(*Frame).document()
      /Users/inanc/grafana/k6b/common/frame.go:322 +0x1ac
  github.com/grafana/xk6-browser/common.(*Frame).waitForSelector()
```

Reply:
See: b4831fe